examples/sig-example.cpp

```diff
@@ -0,0 +1,229 @@
+                 ECDSASignatureFormat sigFormat,
+                 const std::vector<uint8_t> &signature,
+                 const std::vector<uint8_t> &message)
+{
+    std::shared_ptr<MessageVerificationCtx> verifyCtx(nullptr);
+    try {
+        verifyCtx = std::make_shared<ECDSASignaturePublicKeyCtx>(pubKey, digestType, sigFormat);
+    } catch (const MoCOCrWException &e) {
+        std::cerr << "Please check your ECC key. Failure creating context." << std::endl;
+        std::cerr << e.what();
+        exit(EXIT_FAILURE);
+    }
+    try {
+        verifyCtx->verifyMessage(signature, message);
+    } catch (const MoCOCrWException &e) {
+        std::cerr << "Verification failed!" << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    }
+}
+
+std::vector<uint8_t> edDsaSign(const AsymmetricPrivateKey &privKey,
+                               const std::vector<uint8_t> &message)
+{
+    std::shared_ptr<MessageSignatureCtx> signCtx(nullptr);
+    try {
+        signCtx = std::make_shared<EdDSASignaturePrivateKeyCtx>(privKey);
+    } catch (const MoCOCrWException &e) {
+        std::cerr << "Please check your ECC-Ed key. Failure creating context." << std::endl;
+        std::cerr << e.what();
+        exit(EXIT_FAILURE);
+    }
+
+    std::vector<uint8_t> signature;
+    try {
+        signature = signCtx->signMessage(message);
+    } catch (const MoCOCrWException &e) {
+        std::cerr << "Failure occurred during signing." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    }
+    return signature;
+}
+
+void edDsaVerify(const AsymmetricPublicKey &pubKey,
+                 const std::vector<uint8_t> &signature,
+                 const std::vector<uint8_t> &message)
+{
+    std::shared_ptr<MessageVerificationCtx> verifyCtx(nullptr);
+    try {
+        verifyCtx = std::make_shared<EdDSASignaturePublicKeyCtx>(pubKey);
+    } catch (const MoCOCrWException &e) {
+        std::cerr << "Please check your ECC-Ed key. Failure creating context." << std::endl;
+        std::cerr << e.what();
+        exit(EXIT_FAILURE);
+    }
+
+    try {
+        verifyCtx->verifyMessage(signature, message);
+    } catch (const MoCOCrWException &e) {
+        std::cerr << "Verification failed!" << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    }
+}
+
+int main(void) {
+
+    std::vector<uint8_t> message = utility::fromHex("deadbeef");
+
+    /************** RSA signature **************/
+    auto rsaPrivKey = AsymmetricPrivateKey::generateRSA();
+    /* If the default MGF function is used, the digest type for MGF and for PSS-Padding is the same */
+    auto mgf1DigestType = DigestTypes::SHA256;
+    auto rsaSignatureDigestType = DigestTypes::SHA512;
+    auto mgf1 = std::make_shared<MGF1>(MGF1(mgf1DigestType));
+    auto padding = std::make_shared<PSSPadding>(PSSPadding(mgf1, mococrw::Hash::getDigestSize(rsaSignatureDigestType)));
```

Comment:
The only parameter that is unclear from reading the input because its just a number (the salt length) is unnamed. If you want to inline things, inline the MGF but not the salt length because I need to take a look at the function definition to understand what the second parameter is… Better would be to have a variable for the things for which its not clear from the type what their meaning is.